src/couch_replicator_sup.erl

```diff
@@ -13,43 +13,60 @@
 
 -module(couch_replicator_sup).
 -behaviour(supervisor).
--export([start_link/0, init/1]).
+-export([start_link/0, init/1, restart_mdb_listener/0]).
 
 start_link() ->
     supervisor:start_link({local, ?MODULE}, ?MODULE, []).
 
 init(_Args) ->
+    MdbChangesArgs = [
+        <<"_replicator">>,  % DbSuffix
+        couch_replicator,   % Module
+        nil,                % Callback context
+        [skip_ddocs]        % Options
+    ],
     Children = [
-        {couch_replicator_scheduler,
-            {couch_replicator_scheduler, start_link, []},
-            permanent,
-            brutal_kill,
-            worker,
-            [couch_replicator_scheduler]},
         {couch_replicator_scheduler_sup,
             {couch_replicator_scheduler_sup, start_link, []},
             permanent,
             infinity,
             supervisor,
             [couch_replicator_scheduler_sup]},
+        {couch_replicator_scheduler,
+            {couch_replicator_scheduler, start_link, []},
+            permanent,
+            brutal_kill,
+            worker,
+            [couch_replicator_scheduler]},
         {couch_replication_event,
             {gen_event, start_link, [{local, couch_replication}]},
             permanent,
             brutal_kill,
             worker,
             dynamic},
-        {couch_replicator_manager_sup,
-	    {couch_replicator_manager_sup, start_link, []},
-	    permanent,
-            infinity,
-	    supervisor,
-            [couch_replicator_manager_sup]},
-        {couch_replicator_job_sup,
-            {couch_replicator_job_sup, start_link, []},
+        {couch_replicator_clustering,
+            {couch_replicator_clustering, start_link, []},
             permanent,
-            infinity,
-            supervisor,
-            [couch_replicator_job_sup]}
+            brutal_kill,
+            worker,
+            [couch_replicator_clustering]},
+        {couch_replicator,
+            {couch_replicator, start_link, []},
+            transient,
+            brutal_kill,
+            worker,
+            [couch_replicator]},
+        {couch_multidb_changes,
+            {couch_multidb_changes, start_link, MdbChangesArgs},
+            permanent,
+            brutal_kill,
+            worker,
+            [couch_multidb_changes]}
     ],
-    {ok, {{one_for_one,10,1}, Children}}.
+    {ok, {{rest_for_one,10,1}, Children}}.
```

Comment:
Looking more in depth at dependencies:

  * `_scheduler` needs `_scheduler_sup`
  * replications spawned from `_multidb_changes` need `_scheduler`, event manager, and `_clustering`. 
  * If `_clustering` fails we want `_multidb_changes` to restart.  
  * If `_multidb_changes` fails, other children don't need to restart.  
  * If `_scheduler` crashes, we want to `multidb_changes` to restart so it can rescan and re-add the jobs.
  * If `_clustering` crashes we'd have to restart scheduler (this currently won't happen I'll make the update).
  * If event manager crashes, replications running in scheduler need to restart. So it should move above the scheduler (will make this update).